### DISCUSSION TITLE
Question about background tasks

### DISCUSSION THREAD

**Discussion by vminfant** [Neoteroi/BlackSheep#382] [Q&A]
Hi, 

I would like to seek inputs related to background tasks. We run multiple instances of the application in K8s and route the requests using a load balancer. We have defined some clean up routine as a background task in the application. However, each instance of an application will trigger the same task in background which is completely unnecessary. 

Is there any thoughts around how to restrict the task to a specific instance?

**Comment by RobertoPrevato** [comment]
Can you please provide some more information? I understand that your load balancer directs the initial request that triggers a job to a specific instance, and that particular instance handles the job.

Later the clean up routine running as background task, can be picked up by any instances. Your problem is that the clean up routing is picked by each instance?

Did you consider maybe adding a messaging system, for example Rabbit MQ, to write a message about a clean-up task, and use the SDK of the messaging system to have a single consumer handle the task at a time?

**Reply by vminfant** [reply]
hi @RobertoPrevato. Thank you for looking into this. Yes. you are right when the request is forwarded to specific instance from a load balancer, it handles the request and respond back. 

The clean up task is scheduled with a specific interval based on (https://www.neoteroi.dev/blacksheep/background-tasks/#how-to-configure-background-tasks) - but just using background task based on schedule will lead all the instances to run the same task. 

I understand this is not specific to `blacksheep` itself but i wanted to get feedback from community on how they solved such situations.

We haven't explored a messaging system yet but thats really a viable option.
